File: exercises/D205/solution.py
import math

n = input()
print(n + '=', end='')
n = int(n)
m = int(math.sqrt(n))
first = True
for i in range(2, m + 1):
    while n % i == 0:
        if not first:
            print('*', end='')
        print(i, end='')
        n = n // i
        first = False
if n > 1:
    if not first:
        print('*', end='')
    print(n, end='')
print()

File: exercises/D205/solution.java
import java.util.*;

class D205 {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int m = (int) Math.sqrt(n);
        boolean first = true;
        System.out.print(n + "=");
        for (int i = 2; i <= m; i++) {
            while (n % i == 0) {
                if (!first) {
                    System.out.print("*");
                }
                System.out.print(i);
                n /= i;
                first = false;
            }
        }
        if (n > 1) {
            if (!first) {
                System.out.print("*");
            }
            System.out.print(n);
        }
        System.out.println();
    }
}
